### PR TITLE
Set requests timeout

### PR DESCRIPTION
**Pull request by benoitbryon**
Refs #34

### CODE REVIEW

**Comment by benoitbryon** [comment]
Missing tests.

**Comment by benoitbryon** [comment]
@bersace: I reverted (soft at the moment, but I wonder about doing hard reset) commit in master and cherry-picked it here.
Please add tests before we merge master.

As an example, add a test with a extremly small timeout and make sure it actually timeouts. And a test with a big timeout and make sure it does not timeout.

File: pydocusign/test.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def docusign_client_factory(settings=os.environ):
         username=settings['PYDOCUSIGN_TEST_USERNAME'],
         password=settings['PYDOCUSIGN_TEST_PASSWORD'],
         integrator_key=settings['PYDOCUSIGN_TEST_INTEGRATOR_KEY'],
+        **kwargs
```
Comment (benoitbryon) [inline pydocusign/test.py]: Cannot pass "integrator_key" in kwargs. Explicit key/value in kwargs should have priority over (override) settings.
What about something like:

```python
options = {
    username=settings['PYDOCUSIGN_TEST_USERNAME'],
}
options.update(kwargs)
client = DocuSignClient(**kwargs)
```

?

File: tests.py [hunk 2]
```diff
@@ -86,6 +86,12 @@ def test_create_envelope_from_document_request(self):
             '\r\n'
         ))
 
+    def test_timeout(self):
```
Comment (benoitbryon) [inline tests.py]: Add docstring where first line summarizes what you expect (behaviour driven style).
Something like `"""DocuSignClient with (too small) timeout raises exception when issuing requests."""`

File: pydocusign/client.py [hunk 3]
```diff
@@ -47,6 +48,8 @@ def __init__(self,
             self.account_url = '{root}/accounts/{account}'.format(
                 root=self.root_url,
                 account=self.account_id)
+        #: Requests timeout
+        self.http_timeout = timeout
```
Comment (benoitbryon) [inline pydocusign/client.py]: Is there another timeout? Is there a reason for the `http_` prefix?
I think "timeout" is clear enough. We don't care about API calls being HTTP or whatever.
Argument name and attribute name should be consistent: if argument is "timeout", then attribute should be "timeout" as well.
So, imho: `s/http_timeout/timeout/`

File: pydocusign/test.py (hunk 1 above)
Comment (bersace) [inline pydocusign/test.py]: C'est pas une régression. Donc le jour où on voudra le faire, on mettre à jour ce code.

File: pydocusign/client.py [hunk 4]
```diff
@@ -47,6 +48,8 @@ def __init__(self,
             self.account_url = '{root}/accounts/{account}'.format(
                 root=self.root_url,
                 account=self.account_id)
+        #: Requests timeout
```
Comment (benoitbryon) [inline pydocusign/client.py]: Here you can add details about the "timeout" meaning, i.e. you can tell it is used as timeout for HTTP requests to DocuSign API.

Comment (benoitbryon) [inline pydocusign/client.py]: Add a period at the end of the docstring.

See [PEP-257] (http://legacy.python.org/dev/peps/pep-0257/#one-line-docstrings):

> The docstring is a phrase ending in a period.

File: tests.py (hunk 2 above)
Comment (bersace) [inline tests.py]: fait.

File: CHANGELOG [hunk 5]
```diff
@@ -11,6 +11,7 @@ about future releases, check `milestones`_ and :doc:`/about/vision`.
 - Bug #41 - demo/accounts.py cannot be completed automatically: user
   interaction is required to activate account. Added an interactive prompt
   in demo. Removed demo/accounts.py from tests.
+- Feater #34 - add timeout option
```
Comment (benoitbryon) [inline CHANGELOG]: Typo: `s/Feater/Feature`.

"add timeout option" sounds very short to me. As a pydocusign user, in order to understand which changes are introduced, I'd appreciate additional details.
Perhaps something like `Feature #34 - DocuSignClient accepts a ``timeout`` option (defaults to 30 seconds). This timeout is used when performing HTTP requests to DocuSign API.`

Comment (benoitbryon) [inline CHANGELOG]: Note: I do not expect the PR to provide CHANGELOG. I mean, it's ok to provide the CHANGELOG, but it is not blocker. I think CHANGELOG review and update is part of maintainer's job, i.e. I take care of the CHANGELOG on the merge commit.
See also http://tech.novapost.fr/changelog-howto-en.html#maintain-changelog-as-you-merge-pull-requests

**Pull request closed without merging**